### PR TITLE
Return tmp file listing from lambda

### PR DESCRIPTION
**Pull request by sushengloong**
~1. Delete jar file immediately after extracting schedule.~
2. Only download jar file when schedule is not specified through options.
~3. Add new exit error code for exception during schedule extraction.~
4. Return tmp file listing from lambda for troubleshooting

### CODE REVIEW

**Comment by realstraw** [comment]
I like number 2. not sure about number 1, let's chat offline

**Comment by sushengloong** [comment]
@realstraw thanks for the feedback. I have updated the PR to address them.

File: starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala [hunk 1]
```diff
@@ -93,6 +88,8 @@ object SubmitPipeline extends DateTimeFunctions with WaitForIt with DateTimeMapp
 
         (getPeriodFromSchedule(specifiedSchedule), getStartTimeFromSchedule(specifiedSchedule))
       case x =>
+        val jarFile = S3FileHandler.getFileFromS3(opts.jar, opts.baseDir)
+        if (opts.cleanUp) jarFile.deleteOnExit()
```
Comment (talbright) [inline starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala]: Is `deleteOnExit` asynchronous?

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala [hunk 2]
```diff
@@ -58,5 +59,15 @@ class SubmitHandler extends RequestHandler[SubmitRequest, SubmitResponse] with L
 
     SubmitResponse(outString, errString, status, input)
   }
+
+  /**
+   * @return /tmp file listing with size for troubleshooting purposes
+   */
+  private def scanTmpFiles(): String = {
```
Comment (talbright) [inline starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala]: ✅ I was going to suggest something like this...

File: starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala (hunk 1 above)
Comment (talbright) [reply starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala]: Also: why would we ever not want to cleanup?

Comment (realstraw) [reply starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala]: local submit/testing

Comment (realstraw) [reply starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala]: Doc says

```
deleteOnExit()
Requests that the file or directory denoted by this abstract pathname be deleted when the virtual machine terminates.
```

Not sure if async or not

**Comment by sushengloong** [comment]
@realstraw @talbright thanks for the feedback. Is there anything else that I should address in this PR?

**Pull request merged**